### PR TITLE
salt: update cffi resource

### PR DESCRIPTION
**Pull request by carlocab**
See #68463.

### CODE REVIEW

**Comment by fxcoudert** [comment]
`Collecting cffi==1.14.3` 😢

**Comment by carlocab** [comment]
Earlier, though: https://github.com/Homebrew/homebrew-core/pull/68490/checks?check_run_id=1663928195#step:7:4758

`Collecting cffi>=1.14` so 🤷

**Comment by SMillerDev** [comment]
This formula gives issues on every update. Are we sure we still want to ship this?

**Comment by carlocab** [comment]
To be fair at the moment it's giving no greater issues than other formulae that aren't yet ARM-compatible. That said, I don't mind `deprecate!`/`disable!` if it starts to do more than that.

**Comment by carlocab** [comment]
@bayandin do we still need to do an `inreplace` to use a custom `cffi` as mentioned in this comment? https://github.com/Homebrew/homebrew-core/pull/62864#issuecomment-709094985

**Comment by bayandin** [comment]
> @bayandin do we still need to do an `inreplace` to use a custom `cffi` as mentioned in this comment? [#62864 (comment)](https://github.com/Homebrew/homebrew-core/pull/62864#issuecomment-709094985)

We do. 
Currently all versions are listed in [`buildpath/"requirements/static/pkg/py#{xy}/darwin.txt"`](https://github.com/saltstack/salt/blob/v3002.2/requirements/static/pkg/py3.9/darwin.txt). And it has `cffi==1.14.3`.

**Comment by carlocab** [comment]
Ok, cool, thanks. That's not hard. I'll fix it in a moment. Though, if someone has the time to do it now, please feel free.

**Comment by bayandin** [comment]
Feel involved in the creation of this mess with `salt`, sorry folks.

I'll have another look at how we can list all dependencies in tthe formula without pyobjc*

**Comment by carlocab** [comment]
The `inreplace` does get it to build with `cffi 1.14.4`, but now it won't even pass a `salt --version` test:
```
==> brew test --verbose salt
==> FAILED
==> Testing salt
/usr/bin/sandbox-exec -f /private/tmp/homebrew20210112-5661-74v104.sb ruby -W0 -I $LOAD_PATH -- /opt/homebrew/Library/Homebrew/test.rb /opt/homebrew/Library/Taps/homebrew/homebrew-core/Formula/salt.rb --verbose
==> /opt/homebrew/Cellar/salt/3002.2_1/bin/salt --version
WARNING: /opt/homebrew/Cellar/python@3.9/3.9.1_6/Frameworks/Python.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/Python is loading libcrypto in an unsafe way

Error: salt: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: <0> expected but was
<nil>.
```

**Comment by bayandin** [comment]
@carlocab I've listed all the resources, that we use in salt (generated by slightly modified `brew update-python-resources` command, and removed all `pyobjc*`). Hope you don't mind 😄 

Will create a PR for `update-python-resources` to make it easier to support the formula later.

Also, added `pkg-config` dependency to fix the latest failure

**Comment by carlocab** [comment]
That's weird; that wasn't there before I added `depends_on "libffi"`... https://github.com/Homebrew/homebrew-core/runs/1691913764

But, sure, I don't mind the additional changes. Thanks!

**Comment by carlocab** [comment]
Still no luck, sadly

**Comment by bayandin** [comment]
```
      File "/private/tmp/pip-build-env-mghh8j2q/overlay/lib/python3.9/site-packages/cffi/api.py", line 48, in __init__
        import _cffi_backend as backend
    ImportError: dlopen(/private/tmp/pip-build-env-mghh8j2q/overlay/lib/python3.9/site-packages/_cffi_backend.cpython-39-darwin.so, 2): Symbol not found: _ffi_prep_closure
      Referenced from: /private/tmp/pip-build-env-mghh8j2q/overlay/lib/python3.9/site-packages/_cffi_backend.cpython-39-darwin.so
      Expected in: flat namespace
     in /private/tmp/pip-build-env-mghh8j2q/overlay/lib/python3.9/site-packages/_cffi_backend.cpython-39-darwin.so
    Preparing wheel metadata: finished with status 'error'
```


> That's weird; that wasn't there before I added `depends_on "libffi"`... https://github.com/Homebrew/homebrew-core/runs/1691913764


Actually, for most of the formulae, we use system `libffi`, probably we'd like to stick with `uses_from_macos "libffi"` here as well

**Comment by carlocab** [comment]
>Actually, for most of the formulae, we use system `libffi`, probably we'd like to stick with `uses_from_macos "libffi"` here as well

Yes, I would also prefer system `libffi`. I was just trying it to see if it would help...

File: Formula/salt.rb [hunk 1]
```diff
@@ -18,61 +19,254 @@ class Salt < Formula
     sha256 "bbc802eb5097a9f2d02b182215ca73eb37bdd841972e321bcaac219c2b530431" => :mojave
   end
 
+  depends_on "pkg-config" => :build
```
Comment (bayandin) [inline Formula/salt.rb:22]: Ah, forgot to delete this one

**Comment by bayandin** [comment]
I guess that it could be related to the previous huge problem with salt and libcrypto (https://github.com/saltstack/salt/issues/55084) which was fixed by https://github.com/saltstack/salt/pull/56958.

Could anyone check which one `libcrypto` salt is trying to load on Apple Silicon?

**Pull request closed without merging**